internal/receiver/scriptedinputsreceiver/procpipe/commander.go

```diff
@@ -0,0 +1,203 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procpipe
+
+import (
+	"context"
+	_ "embed"
+	"io"
+	"os/exec"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const includePattern = ". \"$(dirname \"$0\")\"/common.sh"
+
+//go:embed scripts/cpu.sh
+var cpuScript string
+
+//go:embed scripts/df.sh
+var dfScript string
+
+//go:embed scripts/ps.sh
+var psScript string
+
+//go:embed scripts/common.sh
+var commonScript string
+
+var scripts = map[string]string{
+	"cpu": cpuScript,
+	"df":  dfScript,
+	"ps":  psScript,
+}
+
+// Commander can start/stop/restart the Agent executable and also watch for a signal
+// for the Agent process to finish.
+type Commander struct {
+	stdout       io.Writer
+	logger       *zap.Logger
+	cmd          *exec.Cmd
+	doneCh       chan struct{}
+	waitCh       chan struct{}
+	execFilePath string
+	args         []string
+	running      int64
+}
+
+func NewCommander(logger *zap.Logger, execFilePath string, stdout io.Writer, args ...string) (*Commander, error) {
+	return &Commander{
+		execFilePath: execFilePath,
+		logger:       logger,
+		args:         args,
+		stdout:       stdout,
+	}, nil
+}
+
+// Start the Agent and begin watching the process.
+// Agent's stdout and stderr are written to a file.
+func (c *Commander) Start(ctx context.Context) error {
+	c.logger.Info("Starting script", zap.String("script", c.execFilePath))
+
+	_, ok := scripts[c.execFilePath]
+	if !ok {
+		c.logger.Error("Unsupported script", zap.Any("script_name", c.execFilePath))
+		return nil
+	}
+
+	c.cmd = exec.CommandContext(ctx, "/bin/sh", c.args...) //nolint:gosec
```

Comment:
This path cannot be guaranteed to exist, like in the collector image for example (`/usr/lib/splunk-otel-collector/agent-bundle/bin/sh`). I think exec walks the path by default but we may want to have a more deterministic lookup process?